tests/hil/tests/ota/test_ota.py

```diff
@@ -0,0 +1,198 @@
+    return artifacts
+
+@pytest.fixture(scope="module")
+async def releases(project, artifacts, tag):
+    releases = dict()
+
+    releases["test_manifest"] = await project.releases.create([artifacts["test_manifest"].id], [], [tag.id], False)
+    releases["test_multiple"] = await project.releases.create([artifacts["test_multiple"].id, artifacts["multi_artifact"].id], [], [tag.id], False)
+    releases["test_blocks"] = await project.releases.create([artifacts["test_blocks"].id], [], [tag.id], False)
+    releases["test_reasons"] = await project.releases.create([artifacts["test_reasons"].id], [], [tag.id], False)
+    yield releases
+
+    # Clean Up
+
+    for r in releases:
+        try:
+            await project.releases.delete(releases[r].id)
+        except:
+            pass
+
+@pytest.fixture(autouse=True, scope="function")
+async def releases_teardown(project, releases):
+    yield
+    release_ids = [r.id for r in releases.values()]
+    updated_release_info = await project.releases.get_all()
+    for r in updated_release_info:
+        if r.id in release_ids:
+            if r.rollout:
+                await project.releases.rollout_set(r.id, False)
+
+##### Tests #####
+
+async def test_manifest(artifacts, board, project, releases):
+    await project.releases.rollout_set(releases["test_manifest"].id, True)
+
+    assert None != board.wait_for_regex_in_line('Manifest received', timeout_s=30)
+
+    assert None != board.wait_for_regex_in_line('Manifest successfully decoded', timeout_s=2)
+
+    # We may need to wait for the second manifest to trigger this line (the first has no rollouts)
+
+    assert None != board.wait_for_regex_in_line('Found main component', timeout_s=30)
+    assert None != board.wait_for_regex_in_line('No absent component found', timeout_s=2)
+
+    # Test all parts of OTA component
+
+    b_info = artifacts["test_manifest"].info["binaryInfo"]
+
+    assert None != board.wait_for_regex_in_line('main.package: main', timeout_s=2)
+    assert None != board.wait_for_regex_in_line(f'main.version: {DUMMY_VER_OLDER}', timeout_s=2)
+    assert None != board.wait_for_regex_in_line(f'main.size: {b_info["size"]}', timeout_s=2)
+    assert None != board.wait_for_regex_in_line('main.is_compressed: 0', timeout_s=2)
+    assert None != board.wait_for_regex_in_line(f'main.hash: {b_info["digests"]["sha256"]["digest"]}', timeout_s=2)
+    assert None != board.wait_for_regex_in_line(f'main.bootloader: {b_info["type"]}', timeout_s=2)
+
+
+async def test_multiple_artifacts(artifacts, board, project, releases):
+    await project.releases.rollout_set(releases["test_multiple"].id, True)
+
+    assert None != board.wait_for_regex_in_line('Manifest received', timeout_s=30)
+    assert None != board.wait_for_regex_in_line('Manifest successfully decoded', timeout_s=2)
+    assert None != board.wait_for_regex_in_line('Found main component', timeout_s=30)
+    assert None != board.wait_for_regex_in_line('Found walrus component', timeout_s=2)
```

Comment:
Any reason to not also check the manifest contents?

Reply:
I've updated to include this. All tests passed, this is ready for review.